nixos/modules/services/databases/mysql.nix

```diff
@@ -274,7 +274,8 @@ in
         serviceConfig = {
           Type = if hasNotify then "notify" else "simple";
           RuntimeDirectory = "mysqld";
-          ExecStart = "${mysql}/bin/mysqld --defaults-file=/etc/my.cnf ${mysqldOptions}";
+          # The last two environment variables are used for starting Galera clusters
+          ExecStart = "${mysql}/bin/mysqld --defaults-file=/etc/my.cnf ${mysqldOptions} ''$_WSREP_NEW_CLUSTER ''$_WSREP_START_POSITION";
```

Comment:
What are these two `'`'s for?

Reply:
> $ can be escaped by prefixing it with ''

[source](https://nixos.org/nix/manual/#ssec-values)

Reply:
This is a `"` string though, not an indented one (as mentioned in the docs you linked). And in any case, $ itself (without any `{` or other special symbol after it) doesn't need to be escaped in any strings. And the way you did it here, literally `''$_WSREP_NEW_CLUSTER` would be passed as an argument (did you actually test this?). You can experiment with escaping in `nix repl`.

In short: Here you can just use `$_WSREP_NEW_CLUSTER $_WSREP_START_POSITION`

Reply:
@Infinisil Fixed and rebased again. Should be good this time.